components/TrackableComponent.js

```diff
@@ -0,0 +1,99 @@
+// Module imports
+import PropTypes from 'prop-types'
+
+
+
+
+
+// Component imports
+import Component from './Component'
+import { pushTrackableEventToDataLayer } from '../helpers'
+
+
+
+
+
+// Component constants
+const gaEventProps = ['action', 'category', 'label', 'value']
+
+
+
+
+
+class TrackableComponent extends Component {
+  /***************************************************************************\
+    Private Methods
+  \***************************************************************************/
+
+  _fireEvent () {
+    const eventData = {}
+
+    for (const key of gaEventProps) {
+      const value = this.props[key]
+      const propName = key.split('').reduce((accumulator, character, index) => `${accumulator}${index === 0 ? character.toUpperCase() : character}`, 'event')
```

Comment:
As a final comment on this method of getting the event prop name, since the event props never actually change, doing this is kinda inefficient. We can just store gaEventProps as a 2d array like this:
```
const gaEventProps = [
  ['action', 'eventAction'],
  ['category', 'eventCategory'],
  ['label', 'eventLabel'],
  ['value', 'eventValue'],
]
```

and get the values of each entry like this:

```
for (const [key, propName] of gaEventProps) {
```